common/persistence/sql/sqlplugin/mysql/cluster_metadata.go

```diff
@@ -46,28 +47,28 @@ cluster_metadata WHERE metadata_partition = ?`
 
 	// ****** CLUSTER_MEMBERSHIP TABLE ******
 	templateUpsertActiveClusterMembership = `REPLACE INTO
-cluster_membership (host_id, rpc_address, rpc_port, role, session_start, last_heartbeat, record_expiry)
-VALUES(?, ?, ?, ?, ?, ?, ?)`
+cluster_membership (membership_partition, host_id, rpc_address, rpc_port, role, session_start, last_heartbeat, record_expiry)
+VALUES(?, ?, ?, ?, ?, ?, ?, ?)`
 
 	templatePruneStaleClusterMembership = `DELETE FROM
 cluster_membership 
-WHERE record_expiry < ? LIMIT ?`
+WHERE membership_partition = ? AND record_expiry < ? LIMIT ?`
 
-	templateGetClusterMembership = `SELECT host_id, rpc_address, rpc_port, role, session_start, last_heartbeat, record_expiry, insertion_order FROM
-cluster_membership`
+	templateGetClusterMembership = `SELECT host_id, rpc_address, rpc_port, role, session_start, last_heartbeat, record_expiry FROM
+cluster_membership WHERE membership_partition = ?`
 
 	// ClusterMembership WHERE Suffixes
 	templateWithRoleSuffix           = ` AND role = ?`
 	templateWithHeartbeatSinceSuffix = ` AND last_heartbeat > ?`
 	templateWithRecordExpirySuffix   = ` AND record_expiry > ?`
 	templateWithRPCAddressSuffix     = ` AND rpc_address = ?`
 	templateWithHostIDSuffix         = ` AND host_id = ?`
-	templateWithSessionStartSuffix   = ` AND session_start > ?`
-	templateWithInsertionOrderSuffix = ` AND insertion_order > ?`
+	templateWithHostIDGreaterSuffix  = ` AND host_id > ?`
+	templateWithSessionStartSuffix   = ` AND session_start >= ?`
 
 	// Generic SELECT Suffixes
-	templateWithLimitSuffix            = ` LIMIT ?`
-	templateWithOrderByInsertionSuffix = ` ORDER BY insertion_order ASC`
+	templateWithLimitSuffix               = ` LIMIT ?`
+	templateWithOrderBySessionStartSuffix = ` ORDER BY membership_partition ASC, host_id ASC`
```

Comment:
no point ordering by membership_partition at this point unless the page token also includes the partition itself, right?

Reply:
You know, I wasn't 100% if its helpful to list that in advance for future-proof sake and since the PrimaryKey is ordered like this. I found some conflicting opinions and need to dig a bit more, but its no-harm to keep afaik.